### ISSUE TITLE
when i use tensorflow implement deepspeech2

### ISSUE THREAD

**Issue by ghost** [NervanaSystems/deepspeech#15]
When I use tensorflow to implement deepspeech2, I use aeon to load the training data, then write it in tfrecord format, and finally use tensorflow to implement it. .

But when the ctc, I used the tensorflow ctc. The code is as follows:

My question is: how to build tensorflow tensors required by tensorflows based on the aeon's labels

    sess = tf.InteractiveSession()
    print 'init net ok'
    logits = net.outputs

    logits = tf.reshape(logits, [batch_size, -1, nout])
    logits = tf.transpose(logits, (1, 0, 2))
    labels = tf.SparseTensor(indices=label_idx, values=label_vals, shape=label_shape)
    loss = ctc_ops.ctc_loss(inputs=logits, labels=labels, sequence_length=seq_len)
    cost = tf.reduce_mean(loss)
    decoded, log_prob = ctc_ops.ctc_greedy_decoder(logits, seq_len)
    ler = tf.reduce_mean(tf.edit_distance(tf.cast(decoded[0], tf.int32),
                                          labels))
    learning_rate = 2e-5
    optimizer = tf.train.MomentumOptimizer(learning_rate,0.9,use_nesterov=True)
    tvars = tf.trainable_variables()
    grads, _ = tf.clip_by_global_norm(tf.gradients(cost, tvars),
                                      400)
    train_op = optimizer.apply_gradients(zip(grads, tvars))
    print 'init train_op'
    n_epoch=1000
     fit(sess,network=net,train_op=train_op,cost=cost,acc=ler,n_epoch=n_epoch,print_freq=5,eval_train=True,
        tensorboard=True,tensorboard_epoch_freq=1)

This is the tensorflow load of a batch of data

    def next_to_batch(self):
        wav_tf, label_tf, seq_len_tf = self.sess.run([self.wav_batch, self.label_b_batch, self.seq_len_batch])
        wav_tf=wav_tf[0]
        label_tf=label_tf[0]
        seq_len_tf=seq_len_tf[0]
        batch_seq_len = []
        batch_labels = []
        for mu in range(batch_size):
            start = int(np.sum(seq_len_tf[:mu]))
            target = label_tf[start:start + seq_len_tf[mu]].tolist()
            batch_labels.append(target)
            batch_seq_len.append(seq_len_tf[mu])
        batch_seq_len = np.asarray(batch_seq_len)
        batch_labels_sparse_tuples = self.sparse_tuple_from(batch_labels)
        wav_tf = np.transpose(wav_tf, (1, 0))
        wav_tf = wav_tf.reshape(batch_size,2998,nbands)
        return wav_tf, batch_seq_len, batch_labels_sparse_tuples

can you help me??
How to build a tensor from the aeON output label???

**Comment by tyler-nervana** [comment]
I'm not familiar enough with the tensorflow deepspeech or ctc code to give much insight here. The details on the buffers that aeon provides for speech transcription can be found here: https://aeon.nervanasys.com/index.html/etl_audio.html#transcription. If you have any specific questions about them, feel free to open an issue at https://github.com/NervanaSystems/aeon.

**Comment by ghost** [comment]
@tyler-nervana ,,neon Deepbirnn layer split_input parameter？
When the second layer rnn time, split_input parameter is set to true ,,, then forward with the former 1152 parameters, backward spread with 1152 parameters ,,, Why is this set?

**Comment by tyler-nervana** [comment]
@guoying1030 I've responded to this question in #16.
